sim/Makefile

```diff
@@ -50,21 +50,28 @@ BLOOP_NAILGUN_PORT ?= 8212
 SCALA_BUILDTOOL_DEPS ?= build.sbt
 
 ifdef ENABLE_BLOOP
-override SCALA_BUILDTOOL_DEPS += $(BLOOP_CONFIG_DIR)/TIMESTAMP
-# Two notes about the bloop invocation:
-# 1) the sed removes a leading {file:<path>} that sometimes needs to be
-#    provided to SBT when a project but not for bloop.
-# 2) Generally, one could could pass '--' to indicate all remaining arguments are
-# 	 destined for the scala Main, however a bug in Bloop's argument parsing causes the
-#    --nailgun-port argument to be lost in this case. Workaround this by prefixing
-#    every main-destined argument with "--args"
-define run_scala_main
-	cd $(base_dir) && bloop --nailgun-port $(BLOOP_NAILGUN_PORT) run $(shell echo $(1) | sed 's/{.*}//') --main $(2)  $(addprefix --args ,$3)
-endef
+  override SCALA_BUILDTOOL_DEPS += $(BLOOP_CONFIG_DIR)/TIMESTAMP
+  # Two notes about the bloop invocation:
+  # 1) the sed removes a leading {file:<path>} that sometimes needs to be
+  #    provided to SBT when a project but not for bloop.
+  # 2) Generally, one could could pass '--' to indicate all remaining arguments are
+  # 	 destined for the scala Main, however a bug in Bloop's argument parsing causes the
+  #    --nailgun-port argument to be lost in this case. Workaround this by prefixing
+  #    every main-destined argument with "--args"
+  define run_scala_main
```

Comment:
Can you explain to me again why we need both a "run_scala_main" and a "run_main".

Reply:
When using the fat jar, the generator and goldengate targets want to call java with the jar instead of the scala build tool.  When when not using the fat jar, we just call the scala build tool all the time.  

We decided to make SBT always use the fat jar and Bloop doesn't support fat jars, so it always calls the scala build tool, Bloop.